front/app/containers/Admin/formBuilder/components/FormFields/index.tsx

```diff
@@ -67,72 +63,167 @@ const FormFields = ({
     watch,
     formState: { errors },
   } = useFormContext();
-  const formCustomFields: IFlatCustomField[] = watch('customFields');
   const locale = useLocale();
+  const formCustomFields: IFlatCustomField[] = watch('customFields');
+
   if (isNilOrError(locale)) {
     return null;
   }
 
+  const FormFieldsContainer = styled(Box)`
+    &:hover {
+      cursor: pointer;
+    }
+  `;
+
+  const isFieldSelected = (fieldId) => {
+    return selectedFieldId === fieldId;
+  };
+
+  const getFieldBackgroundColor = (field) => {
+    if (field.input_type === 'page') {
+      return isFieldSelected(field.id) ? colors.primary : colors.background;
+    }
+    return undefined;
+  };
+
+  const getTitleColor = (field) => {
+    if (field.input_type === 'page' && isFieldSelected(field.id)) {
+      return 'white';
+    }
+    return 'grey800';
+  };
+
+  const getIndexTitleColor = (field) => {
+    if (field.input_type === 'page' && isFieldSelected(field.id)) {
+      return 'white';
+    }
+    return 'teal300';
+  };
+
+  const getIndexForTitle = (formCustomFields, field) => {
+    if (field.input_type === 'page') {
+      // Filter out questions
+      const filteredPages = formCustomFields.filter(
+        (customField) => customField.input_type === 'page'
+      );
+      // Return page title
+      return ` ${filteredPages.indexOf(field) + 1}`;
+    } else {
+      // Filter out pages
+      const filteredQuestion = formCustomFields.filter(
+        (customField) => customField.input_type !== 'page'
+      );
+      // Return question title
+      return ` ${filteredQuestion.indexOf(field) + 1}`;
+    }
+  };
+
   return (
     <DndProvider backend={HTML5Backend}>
-      <Box p="32px" height="100%" overflowY="auto">
+      <Box py="32px" height="100%" overflowY="auto">
         <List key={formCustomFields.length}>
           {formCustomFields.map((field, index) => {
             const hasErrors = !!errors.customFields?.[index];
             let outlineStyle = 'none';
             if (hasErrors) {
               outlineStyle = `1px solid ${colors.error}`;
-            } else if (selectedFieldId === field.id) {
-              outlineStyle = `1px solid ${colors.primary}`;
+            } else if (
+              isFieldSelected(field.id) &&
+              field.input_type !== 'page'
+            ) {
+              outlineStyle = `1px solid ${colors.teal300}`;
             }
             const fieldIdentifier = snakeCase(field.title_multiloc[locale]);
 
             return (
-              <Box
+              <FormFieldsContainer
+                role={'button'}
                 key={field.id}
-                style={{ outline: outlineStyle }}
+                style={{ outline: outlineStyle, outlineOffset: '-1px' }}
```

Comment:
This -1px offset is necessary to render the borders correctly using the current SortableRow component. Maybe this can be changed later when we create a new component without the strict styling, but for now this is needed.